File: packages/typia/src/transformers/features/http/HttpHeadersTransformer.ts
import { HttpHeadersProgrammer } from "../../../programmers/http/HttpHeadersProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.headers",
      write: HttpHeadersProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/HttpHeadersTransformer.go
package http

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type httpHeadersTransformerNamespace struct{}

var HttpHeadersTransformer = httpHeadersTransformerNamespace{}

func (httpHeadersTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "http.headers",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativehttpprogrammers.HttpHeadersProgrammer.Write(x)
		},
	})
}
